doc/cli_guidelines.md

```diff
@@ -0,0 +1,196 @@
+# D-Installer CLI Guidelines
+
+This document defines the syntax for the D-Installer CLI. For more CLI specific aspects like return values, flags, etc, please refer to [clig.dev](https://clig.dev/). Guidelines from clig.dev are agnostic about programming languages and tooling in general, and it can be perfectly used as reference for the D-Installer CLI.
+
+## CLI Syntax
+
+Note: The syntax presented here was previously discussed in this [document](https://gist.github.com/joseivanlopez/808c2be0cf668b4b457fc5d9ec20dc73).
+
+The installation settings are represented by a YAML structure, and the CLI is defined as a set of generic sub-commands and verbs that allow to edit any YAML value in a standard way.
+
+The CLI offers a `config` sub-command for editing the YAML config. The `config` sub-command has verbs for the following actions:
+
+* To load a YAML config file with the values for the installation.
+* To edit any value of the config without loading a new complete file again.
+* To show the current config for the installation.
+* To check the current config.
+
+Moreover, the CLI also offers sub-commands for these actions:
+
+* To ask for the possible values that can be used for some setting (e.g., list of available products).
+* To start and abort the installation.
+* To see the installation status.
+* To answers questions.
+
+### Sub-commands and Verbs
+
+This is the list of all sub-commands and verbs:
+
+~~~
+$ dinstaller install
+Starts the installation.
+
+$ dinstaller abort
+Aborts the installation.
+
+$ dinstaller status
+Prints the current status of the installation process and informs about pending actions (e.g., if there are questions waiting to be answered, if a product is not selected yet, etc).
+
+$ dinstaller watch
+Prints messages from the installation process (e.g., progress, questions, etc).
+
+$ dinstaller config load <file>
+Loads installation config from a YAML file, keeping the rest of the config as it is.
+
+$ dinstaller config show [<key>]
+Prints the current installation config in YAML format. If a <key> is given, then it only prints the content for the given key.
+
+$ dinstaller config set <key>=<value> ...
+Sets a config value for the given key.
```

Comment:
in general I like this structure as it looks like man page or help. But for both case I think we should mention nested values separated dot here and not few lines below under yaml config section.